web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/TraceContextCorrelation.java

```diff
@@ -411,4 +416,19 @@ public static String generateChildDependencyTraceparent() {
 
         return null;
     }
+
+    /**
+     * This is helper method to convert traceparent (W3C) format to AI legacy format for supportability
+     * @param traceparent
+     * @return legacy format traceparent
+     */
+    public static String createChildIdFromTraceparentString(String traceparent) {
+        System.out.println(traceparent + "-------");
+        assert traceparent != null;
+
+        String[] traceparentArr = traceparent.split("-");
+        assert traceparentArr.length == 4;
+
+        return "|" + traceparentArr[0] + "." + traceparentArr[1] + ".";
```

Comment:
this cannot work. traceparentArr[0]  is probably version - 00

Reply:
Yes it should be return "|" + traceparentArr[1] + "." + traceparentArr[2] + "."; 
1 - > traceId
2 - > spanId.